File: pr2/src/main/java/Task28.java
public class Task28 {
    public static double func(double x, double y) {
        // (2*X^3-4*X^2+X+1)/(9*Y^3+Y+4) + 3*Y^2+5*Y
        return (2 * Math.pow(x, 3) - 4 * Math.pow(x, 2) + x +1) / (9 * Math.pow(y, 3) + y + 4) + 3 * Math.pow(y, 2) + 5 * y;
    }
}

File: pr3/task28.js
module.exports = function task28() {
  const arr = [];
  const step = 1.2;

  for (let x = -12; x <= 5; x += step) {
    //  z=sin(x)-5cos(x-2) 
    const z = Math.sin(x) - 5 * Math.cos(x - 2);
    
    if (z > 0) {
      arr.push(z);
    }
  }

  return arr;
}
